src/rgw/rgw_sal_motr.cc

```diff
@@ -2649,6 +2689,75 @@ int MotrAtomicWriter::process(bufferlist&& data, uint64_t offset)
   return this->write();
 }
 
+int MotrObject::update_null_reference(const DoutPrefixProvider *dpp, rgw_bucket_dir_entry& ent)
+{
+  int rc = 0;
+  string tenant_bkt_name = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
+
+  string bucket_index_iname = "motr.rgw.bucket.index." + tenant_bkt_name;
+  std::string null_ref_key = this->get_name() + "/null";
+  bufferlist::const_iterator iter;
+  bufferlist bl, bl_null_idx_val;
+  rgw_bucket_dir_entry current_null_key_ref;
+  current_null_key_ref.key.name = this->get_name();
+  current_null_key_ref.key.instance = this->get_instance();
+  current_null_key_ref.encode(bl_null_idx_val);
+
+  int motr_rc = this->store->do_idx_op_by_name(bucket_index_iname,
+                            M0_IC_GET, null_ref_key, bl);
+  ldpp_dout(dpp, 20) <<__func__<< "GET null index entry, rc : " << motr_rc << dendl;
+
+  if(motr_rc == 0)
+  {
+    rgw_bucket_dir_entry null_key;
+    bufferlist& blr = bl;
+    iter = blr.cbegin();
+    null_key.decode(iter);
+    std::string prev_null_obj_key = null_key.key.name + '[' + null_key.key.instance + ']';
+    // delete prev null object entry
+    // clear the bufferlist -> obj1/null = {}
+    bl.clear();
+    rc = this->store->do_idx_op_by_name(bucket_index_iname,
+                M0_IC_DEL, prev_null_obj_key, bl);
+    if(rc < 0)
+    { 
+      ldpp_dout(dpp, 0) <<__func__<< " ERROR: Unable to null object key" << dendl;
```

Comment:
i think 'fetch' word in missing, check log statement

Reply:
done